### PR TITLE
Harden `windows_get_shell`

### PR DESCRIPTION
**Pull request by dra27**
In Windows native containers, the `Filename.chop_suffix` call in `windows_get_shell` was occasionally failing. `OpamStubs.getProcessAncestry` can return empty strings. `windows_get_shell` now explictly handles the empty string case and `.exe` is now _optionally_ stripped.

### CODE REVIEW

**Review by rjbou — APPROVED** [review]

**Comment by rjbou** [comment]
related to https://github.com/ocaml/opam/issues/5644#issuecomment-1697443468

File: master_changes.md [hunk 1]
```diff
@@ -105,6 +105,7 @@ users)
 ## Internal
 
 ## Internal: Windows
+  * Fix sporadic crash in shell detection (seen in native containers) [#5714 @dra27]
```
Comment (rjbou) [inline master_changes.md:108]: note for dev: Missing api changelog

Comment (dra27) [reply master_changes.md]: Done

**Comment by dra27** [comment]
I pushed two commits fixing errors in the C stub for determining the shell, one of which was heinous (the pointer into the array would end up pointing at the _old_ array following `realloc`)

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**